Domain/Submission/src/main/java/tw/waterball/judgegirl/submission/domain/usecases/RejudgeAllSubmissionsUseCase.java

```diff
@@ -0,0 +1,35 @@
+package tw.waterball.judgegirl.submission.domain.usecases;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import tw.waterball.judgegirl.primitives.submission.Submission;
+import tw.waterball.judgegirl.submission.domain.repositories.SubmissionRepository;
+import tw.waterball.judgegirl.submission.domain.usecases.dto.SubmissionQueryParams;
+
+import javax.inject.Named;
+import java.util.List;
+
+/**
+ * Find a best record submission given a list of submission ids
+ *
+ * @author - c11037at@gmail.com (Snowmancc)
+ */
+@Named
+@AllArgsConstructor
+public class RejudgeAllSubmissionsUseCase {
+    private final SubmissionRepository submissionRepository;
+    private final RejudgeSubmissionUseCase rejudgeSubmissionUseCase;
+
+    public void execute(SubmissionQueryParams queryParams) {
+        List<Submission> submissions = submissionRepository.query(queryParams);
+
+        for (Submission submission : submissions) {
+            rejudgeSubmissionUseCase.execute(submission);
+        }
+    }
+
+    @Data
+    public static class Request {
+        public int problemId;
+    }
```

Comment:
I don't think this class is necessary

why not just use `RejudgeSubmissionsUseCase` to handle it?
(remember to rename `RejudgeSubmissionUseCase` to `RejudgeSubmissionsUseCase`)